### DISCUSSION TITLE
Car state suddenly showing as unavailable, model is wrong and other data issues.

### DISCUSSION THREAD

**Discussion by CrispinPZA** [teslamate-org/teslamate#3484] [Q&A] [answered]
I've been using TeslaMate for a while and it's never missed a beat. 
All of a sudden the data seems pretty messed up, like it's someone else's car.  

- it's showing the car state as "unavailable"
- the model is showing as 74D
- the mileage is not my mileage
- battery range is not the same as the Tesla app is showing.

- Running in docker, just pulled latest and problem persists. 
- Late 2022 Model Y, UK, Leased

Is this happening for anyone else? 

Really odd.

**Comment by CrispinPZA** [comment]
I sighed out and back in again with refreshed tokens and it works ok now. 

Did my stuff expire?

**Comment by CrispinPZA** [comment]
naaa, something is messed up. 
None of the charging details are available now. 
I've unplugged the car and plugged it back in but nothing shows up in mqtt.

**Comment by brianmay** [comment]
Does Teslamate's logs show anything useful?

**Reply by CrispinPZA** [reply]
hmm, not sure. Trying to find the logs. Help with a clue please?

**Reply by CrispinPZA** [reply]
FOund them, Seeing this in the logs. 


`2023-11-22 21:51:30.325 [info] Starting logger for 'Tina'
2023-11-22 21:51:30.494 car_id=1 [info] Start / :online
2023-11-22 21:51:30.501 [error] GenStateMachine :"1" terminating
** (MatchError) no match of right hand side value: {:error, #Ecto.Changeset<action: :insert, changes: %{date: ~U[2023-11-22 21:51:30.455000Z], power: 0, odometer: 31673.085505, battery_heater: false, driver_temp_setting: Decimal.new("22.0"), fan_status: 0, inside_temp: Decimal.new("10.0"), is_climate_on: false, is_front_defroster_on: false, is_rear_defroster_on: false, outside_temp: Decimal.new("9.0"), passenger_temp_setting: Decimal.new("22.0"), tpms_pressure_fl: Decimal.new("3.075"), tpms_pressure_fr: Decimal.new("3.125"), tpms_pressure_rl: Decimal.new("3.075"), tpms_pressure_rr: Decimal.new("3.175"), battery_heater_on: false, battery_level: 41, usable_battery_level: 41, ideal_battery_range_km: Decimal.new("212.47"), rated_battery_range_km: Decimal.new("212.47"), est_battery_range_km: Decimal.new("157.8")}, errors: [latitude: {"can't be blank", [validation: :required]}, longitude: {"can't be blank", [validation: :required]}], data: #TeslaMate.Log.Position<>, valid?: false>}
    (teslamate 1.27.3) lib/teslamate/vehicles/vehicle.ex:721: anonymous fn/4 in TeslaMate.Vehicles.Vehicle.handle_event/4
    (ecto_sql 3.10.1) lib/ecto/adapters/sql.ex:1203: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4
    (db_connection 2.5.0) lib/db_connection.ex:1630: DBConnection.run_transaction/4
    (teslamate 1.27.3) lib/teslamate/vehicles/vehicle.ex:713: TeslaMate.Vehicles.Vehicle.handle_event/4
    (stdlib 5.0.1) gen_statem.erl:1379: :gen_statem.loop_state_callback/11
    (stdlib 5.0.1) proc_lib.erl:241: :proc_lib.init_p_do_apply/3
2023-11-22 21:51:30.645 car_id=1 [info] Start / :online
2023-11-22 21:51:30.652 [error] GenStateMachine :"1" terminating
** (MatchError) no match of right hand side value: {:error, #Ecto.Changeset<action: :insert, changes: %{date: ~U[2023-11-22 21:51:30.605000Z], power: 0, odometer: 31673.085505, battery_heater: false, driver_temp_setting: Decimal.new("22.0"), fan_status: 0, inside_temp: Decimal.new("10.0"), is_climate_on: false, is_front_defroster_on: false, is_rear_defroster_on: false, outside_temp: Decimal.new("9.0"), passenger_temp_setting: Decimal.new("22.0"), tpms_pressure_fl: Decimal.new("3.075"), tpms_pressure_fr: Decimal.new("3.125"), tpms_pressure_rl: Decimal.new("3.075"), tpms_pressure_rr: Decimal.new("3.175"), battery_heater_on: false, battery_level: 41, usable_battery_level: 41, ideal_battery_range_km: Decimal.new("212.47"), rated_battery_range_km: Decimal.new("212.47"), est_battery_range_km: Decimal.new("157.8")}, errors: [latitude: {"can't be blank", [validation: :required]}, longitude: {"can't be blank", [validation: :required]}], data: #TeslaMate.Log.Position<>, valid?: false>}
    (teslamate 1.27.3) lib/teslamate/vehicles/vehicle.ex:721: anonymous fn/4 in TeslaMate.Vehicles.Vehicle.handle_event/4
    (ecto_sql 3.10.1) lib/ecto/adapters/sql.ex:1203: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4
    (db_connection 2.5.0) lib/db_connection.ex:1630: DBConnection.run_transaction/4
    (teslamate 1.27.3) lib/teslamate/vehicles/vehicle.ex:713: TeslaMate.Vehicles.Vehicle.handle_event/4
    (stdlib 5.0.1) gen_statem.erl:1379: :gen_statem.loop_state_callback/11
    (stdlib 5.0.1) proc_lib.erl:241: :proc_lib.init_p_do_apply/3
`

**Reply by CrispinPZA** [reply]
Looks like there is an issue with the position and the code is falling over?

**Reply by brianmay** [reply, marked as answer]
That looks like you are still running an old version of Teslamate, that hasn't been updated with the endpoint fix. Please make sure you are running the latest released version and that should go away,

**Reply by CrispinPZA** [reply]
Boom! Thank you!

I had pulled a new image but get I got something wrong. Deleted the container and image and tried again. 

Thanks for your help!
